### PR TITLE
Post articles comments

### PR DESCRIPTION
**Pull request by Iblis1871**
post comments for review

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -68,6 +118,34 @@ describe("PATCH api/articles", () => {
       });
   });
 });
+describe("POST /api/articles/:article_id/comments", () => {
+  test("status:200, responds with a added comment ONE", async () => {
```
Comment (karlaferry) [inline __tests__/app.test.js:122]: Your two 200 status code tests for this endpoint looks like they're testing the same thing.

File: models/comments.models.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+const db = require("../db/connection");
+const articles = require("../db/data/test-data/comments");
+
+exports.newsComments = async (article_id) => {
+  const comments = await db.query(
+    `SELECT * FROM comments WHERE article_id = $1`,
+    [article_id]
+  );
+  return comments.rows[0];
```
Comment (karlaferry) [inline models/comments.models.js:9]: Should return `comments.rows` 😄

**Review by karlaferry — CHANGES_REQUESTED** [review]
Please see comments and resubmit for approval. Thank you 😄

File: __tests__/app.test.js [hunk 3]
```diff
@@ -31,54 +31,45 @@ describe("GET api/articles", () => {
...
+      votes: 100,
+      article_id: 1,
+      comment_count: "11",
+    });
   });
 });
 describe("GET api/articles/:article_id/comments", () => {
-  test("status:200, responds the comments for a given article_id, ONE", () => {
-    return request(app)
+  test("status:200, responds the comments for a given article_id, ONE", async () => {
```
Comment (Rosem10) [inline __tests__/app.test.js:49]: Your test description shouldn't refer to one specific article - it should reflect that it works for any id (just take the "ONE" out, please.)

File: __tests__/app.test.js [hunk 4]
```diff
@@ -31,54 +31,45 @@ describe("GET api/articles", () => {
...
-          comment_id: 14,
-          body: "What do you see? I have no idea where this will lead us. This place I speak of, is known as the Black Lodge.",
-          article_id: 5,
-          author: "icellusedkars",
-          votes: 16,
-          created_at: "2020-06-09T05:00:00.000Z",
-        });
-      });
+
+  test("status:200, responds the comments for a given article_id, ARRAY", async () => {
```
Comment (Rosem10) [inline __tests__/app.test.js:67]: Again, could you please just reword this to be a bit clearer? I appreciate you might be adding notes for yourself - but "responds with an array of comments for..." is way nicer than having that ARRAY on the end.

File: __tests__/app.test.js [hunk 5]
```diff
@@ -184,73 +162,47 @@ describe("GET api/users", () => {
...
       .patch("/api/articles/999")
       .send({ inc_votes: 1 })
-      .expect(404)
-      .then((res) => {
-        expect(res.statusCode).toBe(404);
-        expect(res.body.msg).toBe("path not found!");
-      });
+      .expect(404);
+    expect(res.statusCode).toBe(404);
+    expect(res.body.msg).toBe("path not found!");
```
Comment (Rosem10) [inline __tests__/app.test.js:179]: This one isn't "path not found", as the path of /articles/articleId is one that exists. It's rather that the article is not found.

File: __tests__/app.test.js [hunk 6]
```diff
@@ -184,73 +162,47 @@ describe("GET api/users", () => {
...
-      .expect(400)
-      .then((res) => {
-        console.log(res.body);
-        expect(res.body.msg).toBe("bad request!");
-      });
+  test("status:400, bad request", async () => {
+    const res = await request(app).get("/api/articles/99l99").expect(400);
+    expect(res.body.msg).toBe("bad request!");
   });
-  test("status:404, responds with a path not found ID DOES NOT EXIST", () => {
```
Comment (Rosem10) [inline __tests__/app.test.js]: The way you have laid these tests out makes them hard to follow. It would be useful to separate the different endpoints out in a describe block that makes it clear what endpoint they are testing for. It might be easier to put the relevant error tests with their matching "happy path" tests. At the moment, there's nothing to denote that we are switching from testing for our get request to our patch request.

File: __tests__/app.test.js [hunk 7]
```diff
@@ -184,73 +162,47 @@ describe("GET api/users", () => {
...
-        console.log(res.body);
-        expect(res.body.msg).toBe("path not found!");
-      });
+  test("status:404, path not found", async () => {
+    const res = await request(app).get("/api/articles/1/stnemmoc").expect(404);
+    expect(res.body.msg).toBe("path not found!");
   });
-  test("status:404, should return 400 given an empty object", () => {
-    return request(app)
+  test("status:404, should return 400 given an empty object", async () => {
```
Comment (Rosem10) [inline __tests__/app.test.js:201]: This says both 404 and 400 - please pick one.

**Review by Rosem10 — CHANGES_REQUESTED** [review]
Couple of things to change please!

File: __tests__/app.test.js [hunk 8]
```diff
@@ -31,15 +31,56 @@ describe("GET api/articles", () => {
...
+      );
+    });
+  });
+
+  test("status:200, responds the comments for a given article_id", async () => {
+    const { body } = await request(app)
+      .get("/api/articles/3/comments")
+      .expect(200);
+    const { comments } = body;
+    expect(comments).toBeInstanceOf(Array);
```
Comment (karlaferry) [inline __tests__/app.test.js:72]: You could move this expect statement to the previous test block. No need to separate it from the first one.

File: controllers/comments.controllers.js [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+  const { article_id } = req.params;
+  newsComments(article_id)
+    .then((comments) => {
+      res.status(200).send({ comments });
+    })
+    .catch(next);
+};
+
+exports.postCommentsById = (req, res, next) => {
+  commentsPost(req.params.article_id, req.body)
```
Comment (karlaferry) [inline controllers/comments.controllers.js:13]: Try to be more consistent with destructuring. Line 4 you destructured article_id but not one line 13.

**Review by karlaferry — APPROVED** [review]
Good! Just left a couple more comments then I think you should be ok to merge after. 😄
